### ISSUE TITLE
[bug] gitignored files shown after folder is opened via find

### ISSUE THREAD

**Issue by alex-courtis** [kyazdani42/nvim-tree.lua#969]
**OS**
`Arch Linux 5.16.8-arch1-1`

**Neovim version**
```
NVIM v0.6.1
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by builduser

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

**Nvim-tree version**
0f7da14

**Describe the bug**
When finding a file inside a closed directory, the directory is opened and the file caret set, hovewer the gitignored files are visible.

Upon refreshing the ignored files are hidden, however the caret is then on the incorrect file.

**To Reproduce**
empty nvim configuration, no plugins othen than nvim-tree
```
NvimTreeOpen
:e dir/file
NvimTreeFindFile
```

[![asciicast](https://asciinema.org/a/468650.svg)](https://asciinema.org/a/468650)

**Expected behavior**
gitignored files should not be visible.

**Comment by alex-courtis** [comment]
@kyazdani42 apologies for the spam of issues. I do intend to fix and PR these when time permits.

**Comment by kyazdani42** [comment]
I spot the issue in the code:
the find file action calls `reload_node_status` after exploring with an empty git table.
This code should be rewritten to take into account a full git reload before exploring, and taking into account the asynchrony + the recursion.

**Comment by alex-courtis** [comment]
Still an issue at 3f4ed9b

**Comment by alex-courtis** [comment]
NvimTreeFindFile case resolved at d34ea42 - great work, thank you.

**Comment by alex-courtis** [comment]
Also working for the `update_focused_file.enabled` case.
